### ISSUE TITLE
When training with GRPO, a warning has appeared: None of the inputs have requires_grad=True. Gradients will be None

### ISSUE THREAD

**Issue by ChenJueCharles** [FareedKhan-dev/train-deepseek-r1#8]
Although it does not affect model training, the following warnings or error occur:

1.`use_cache=True` is incompatible with gradient checkpointing. Setting `use_cache=False`.
![Image](https://github.com/user-attachments/assets/7a3eaef1-162a-45ee-acb7-691538f1333a)
2.None of the inputs have requires_grad=True. Gradients will be None
![Image](https://github.com/user-attachments/assets/2153e9e2-8bdb-4247-9c26-7626e8472acd)
3.Error during comparison
![Image](https://github.com/user-attachments/assets/f4784d63-21f0-413f-97ff-54f68cc6146b)

I sincerely hope that the author has time to answer.
